packages/edit-site/src/components/editor/index.js

```diff
@@ -138,9 +138,18 @@ export default function Editor() {
 	);
 	const isReady = editedPostType !== undefined && editedPostId !== undefined;
 
+	let titleBreadcrumb;
+	if ( isReady && editedPost ) {
+		const type =
+			editedPostType === 'wp_template'
+				? __( 'Template' )
+				: __( 'Template Part' );
+		titleBreadcrumb = `${ editedPost.title?.rendered } ‹ ${ type } ‹ `;
+	}
```

Comment:
Nit: I think can just construct the `title` string here for simplicity (`useMemo` is probably not needed):

```suggestion
	const title = useMemo( () => {
		const type =
			editedPostType === 'wp_template'
				? __( 'Template' )
				: __( 'Template Part' );
		return `${ editedPost.title?.rendered } ‹ ${ type } ‹ ${ __( 'Editor (beta)' ) }`
	}, [ editedPost.title, editedPostType ] );
```

Then later we can just do:

```js
useTitle( isReady && title );
```

Reply:
I was thinking that there might be instances where `editedPost` was not set so title would then just need to be `Editor (beta)` but I did some more testing and this doesn't seem to ever be the case, so have simplified as you suggested, but without the useMemo  - this isn't a calculation heavy bit of logic so I agree that this probably isn't needed.